### PR TITLE
Reduce hacks needed for bundle format CI.

### PR DESCRIPTION
**Pull request by markusthoemmes**
- Dropped the need to run with a hacked version of OPM by specifying the internal registry as an insecure registry to make podman do "the right thing" automatically.
- Dropped the need to run the index image as a root by specifying the needed access in the image.
- Checked in the Dockerfile to produce the image here to make rebuilding easier if ever needed.

### CODE REVIEW

File: olm-catalog/serverless-operator/index.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# Create a barebones /etc/nsswitch.conf file.
+# Required to avoid "Error: open /etc/nsswitch.conf: permission denied".
+RUN [ ! -e /etc/nsswitch.conf ] && echo 'hosts: files dns' > /etc/nsswitch.conf
+
+# Run as nobody and create a properly owned HOME directory for podman to be able to write
+# its config files. Also set that directory as a workdir to be able to create more files.
+ENV USER=nobody
+ENV HOME /home/$USER
+RUN mkdir -p $HOME && chgrp -R 0 $HOME && chmod -R g=u $HOME
```
Comment (cardil) [inline olm-catalog/serverless-operator/index.Dockerfile:18]:
```suggestion
RUN mkdir -p $HOME && chown -R nobody:nobody $HOME && chmod -R g=u $HOME
```

File: olm-catalog/serverless-operator/index.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+RUN echo "http://nl.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories && \
+    echo "http://nl.alpinelinux.org/alpine/edge/community" >> /etc/apk/repositories && \
+    apk update && apk add podman
+
+# Add internal registry to insecure registries.
+RUN printf "[registries.insecure]\nregistries = ['image-registry.openshift-image-registry.svc:5000']" > /etc/containers/registries.conf
+
+# Create a barebones /etc/nsswitch.conf file.
+# Required to avoid "Error: open /etc/nsswitch.conf: permission denied".
+RUN [ ! -e /etc/nsswitch.conf ] && echo 'hosts: files dns' > /etc/nsswitch.conf
```
Comment (cardil) [inline olm-catalog/serverless-operator/index.Dockerfile:12]:
```suggestion
RUN [ ! -e /etc/nsswitch.conf ] && echo 'hosts: files dns' > /etc/nsswitch.conf
RUN chgrp nobody /etc/nsswitch.conf && chmod g=u /etc/nsswitch.conf
```

File: hack/lib/catalogsource.bash [hunk 3]
```diff
@@ -73,11 +68,10 @@ spec:
         - /bin/sh
         - -c
         - |-
-          podman login -u $pull_user -p $token --tls-verify=false image-registry.openshift-image-registry.svc:5000
-          mkdir -p /database && \
-          /bin/opm registry add                         -d /database/index.db --mode=replaces -b docker.io/markusthoemmes/serverless-bundle:1.7.2
-          /bin/opm registry add --container-tool=podman -d /database/index.db --mode=replaces -b image-registry.openshift-image-registry.svc:5000/$OLM_NAMESPACE/serverless-bundle && \
-          /bin/opm registry serve -d /database/index.db -p 50051
+          podman login -u $pull_user -p $token image-registry.openshift-image-registry.svc:5000 && \
+          /bin/opm registry add                         -d index.db --mode=replaces -b docker.io/markusthoemmes/serverless-bundle:1.7.2 && \
```
Comment (cardil) [inline hack/lib/catalogsource.bash:72]: `docker.io/markusthoemmes/serverless-bundle` this is to be changed in different PR?

Comment (markusthoemmes) [reply hack/lib/catalogsource.bash:72]: No that's a completely separate image. That'll stay there I assume for 1.8.0.

File: olm-catalog/serverless-operator/index.Dockerfile (hunk 2 above)
Comment (cardil) [reply olm-catalog/serverless-operator/index.Dockerfile:12]: This might be unnecessary if the code only reads the nsswitch file

File: olm-catalog/serverless-operator/index.Dockerfile (hunk 1 above)
Comment (markusthoemmes) [reply olm-catalog/serverless-operator/index.Dockerfile:18]: This doesn't actually work when run on Openshift.

File: olm-catalog/serverless-operator/index.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+RUN echo "http://nl.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories && \
+    echo "http://nl.alpinelinux.org/alpine/edge/community" >> /etc/apk/repositories && \
+    apk update && apk add podman
+
+# Add internal registry to insecure registries.
+RUN printf "[registries.insecure]\nregistries = ['image-registry.openshift-image-registry.svc:5000']" > /etc/containers/registries.conf
+
+# Make /etc accessable by the entire root group.
+# Required to avoid "Error: open /etc/nsswitch.conf: permission denied".
+RUN chgrp -R 0 /etc && chmod -R g=u /etc
```
[the remark this answers is not in the document]
Comment (markusthoemmes) [reply olm-catalog/serverless-operator/index.Dockerfile]: Done.

File: olm-catalog/serverless-operator/index.Dockerfile (hunk 1 above)
Comment (cardil) [reply olm-catalog/serverless-operator/index.Dockerfile:18]:
```suggestion
RUN addgroup $USER root
RUN mkdir -p $HOME && chown -R $USER:root $HOME && chmod -R g=u $HOME
```
What about this. I've tested that on OCP. It also works everywhere, as the permissions to home dir are correct, and thous reducing confusion.

File: hack/lib/catalogsource.bash (hunk 3 above)
Comment (cardil) [reply hack/lib/catalogsource.bash:72]: I know that other image. I'm asking should that be moved to openshift-knative org on quay.io as well?

File: olm-catalog/serverless-operator/index.Dockerfile (hunk 1 above)
Comment (markusthoemmes) [reply olm-catalog/serverless-operator/index.Dockerfile]: Done.
